crates/alkahest-renderer/src/ecs/utility.rs

```diff
@@ -174,6 +234,69 @@ impl Default for Route {
     }
 }
 
+impl Route {
+    pub fn get_command(&self, scene: &Scene, entity: Entity) -> String {
+        let mut command = String::from("route");
+        if let Some(hash) = self.activity_hash.as_ref() {
+            command = format!("{} hash {}", command, hash.0);
+        }
+        if let Some(children) = scene.entity(entity).get::<Children>() {
+            for child_ent in &children.0 {
+                let Some(pos) = scene.entity(*child_ent).get::<Transform>() else {
+                    return String::from("ERR");
+                };
+                let Some(node) = scene.entity(*child_ent).get::<RouteNode>() else {
+                    return String::from("ERR");
+                };
+                let Some(label) = scene.entity(*child_ent).get::<Label>() else {
+                    return String::from("ERR");
+                };
+                command = format!(
+                    "{} node {} {} {}{}{}{}",
+                    command,
+                    pos.translation[0],
+                    pos.translation[1],
+                    pos.translation[2],
+                    if node.is_teleport { " tp" } else { "" },
+                    node.map_hash
+                        .map_or(String::new(), |h| { format!(" hash {}", h.0) }),
+                    if !label.default {
+                        format!(
+                            " label {}",
+                            label.label.replace('\\', r"\\").replace(' ', r"\s")
+                        )
+                    } else {
+                        String::new()
+                    }
+                );
+            }
+        }
+        command
```

Comment:
Use the `std::write` macros instead of format chaining
See https://doc.rust-lang.org/stable/std/macro.write.html